### PR TITLE
Add support for inline conditionals

### PR DESCRIPTION
**Pull request by munkius**
I spotted conditionals only work when the statements are placed in separate nodes (paragraphs in practice). This PR makes it possible inline conditionals work as well.

### CODE REVIEW

File: lib/sablon/processor.rb [hunk 1]
```diff
@@ -49,6 +49,20 @@ def fill_empty_table_cells(xml_node)
...
     class Block < Struct.new(:start_field, :end_field)
+
+      def initialize(start_field, end_field)
+        super(start_field, end_field)
+
+        start_field_parent = self.class.parent(start_field).first
+        end_field_parent   = self.class.parent(end_field).first
+
+        @start_node, @end_node = if start_field_parent == end_field_parent
+                                   [start_field.node, end_field.node]
```
Comment (senny) [inline lib/sablon/processor.rb]: Does this work with `ComplexField`s? They don't consist of a single `node`.

**Comment by senny** [comment]
As you reported conditionals (and other blocks like looks) don't work inline. Would be awesome to get it working! A solution should take `ComplexField`s into account and work for loops and conditionals equally.

Would you mind adding some more integration tests to lock in the behavior and prevent against regressions? I'm thinking about:
* Adding an inline conditional in the cv_sample template
* Adding an inline loop in the cv_sample template
* Add a test for inline blocks and ComplexFields to `test/processor_test.rb`

Thank you for working on this :yellow_heart: Let me know if you have any questions.

**Comment by munkius** [comment]
Hi @senny I'm trying to think of an example in which if inline loops would be of (realistic, not theoretical) use, can't think of any tbh. You? I'll try to find the time to add some more tests for the inline conditional :)

**Comment by senny** [comment]
I can't think of a non-artificial example for inline loops. However given they are both implemented based on the same building blocks one would expect that they behave the same. In terms of testing I'd just loop over a list and print every item separated by a space.

**Comment by senny** [comment]
This behavior was required for some of my use-cases as well. I took over your PR and added the missing pieces.

@munkius thank you very much for your work :yellow_heart:

**Comment by senny** [comment]
Just released `0.0.16` with support for inline conditionals.

**Comment by munkius** [comment]
Awesome dude, thanks 👊

**Pull request closed without merging**